src/clojure_sandbox/4clojure.clj

```diff
@@ -0,0 +1,337 @@
+(ns clojure-sandbox.4clojure
+  (:require [clojure.string :as str]))
+
+;; 21. Write a function which returns the Nth element from a sequence.
+(defn nth-element [[current & remaining :as items] position]
+  (when items
+    (if (pos? position)
+      (recur remaining (dec position))
+      current)))
+
+(nth-element [] 1)
+(nth-element [nil 1 2 3 4] 2)
+(nth-element [0 1 2 3] 5)
+
+;; 22. Write a function which returns the total number of elements in a sequence.
+(defn count-total [init-sequence]
+  (loop [count 0, remaining init-sequence]
+    (if (seq remaining)
+      (recur (inc count) (rest remaining))
+      count)))
+
+(count-total [])
+(count-total [1 2 3])
+(count-total [1 2 3 4 5])
+
+
+;; 23. Write a function which reverses a sequence.
+(defn reverse-sequence [init-sequence]
+  (loop [remaining init-sequence, reversed []]
+    (if (seq remaining)
+      (recur (rest remaining) (cons (first remaining) reversed))
+      reversed)))
+
+(reverse-sequence [1 2 3 4])
```

Comment:
Put these in comment blocks to avoid evaluating them when the file is evaluated. See: https://calva.io/rich-comments/